### ISSUE TITLE
broken code after cargo clippy --fix

### ISSUE THREAD

**Issue by jgoerzen** [rust-lang/rust-clippy#9363]
### Summary

After upgrading Rust from 1.61.0 to 1.63.0, new Clippy errors started to occur.

The first was in this code:

```
warning: casting the result of `i8::abs()` to usize
    --> core/src/suggestions.rs:3483:5
     |
3483 |     (params.rr_y1 - params.rr_y0).abs() as usize
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `params.rr_y1 - params.rr_y0.unsigned_abs()`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_abs_to_unsigned
```

It produced this:

```
warning: failed to automatically apply fixes suggested by rustc to crate `core`

after fixes were automatically applied the compiler reported errors within these files:

  * core/src/suggestions.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see 
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0308]: mismatched types
    --> core/src/suggestions.rs:3483:20
     |
3483 |     params.rr_y1 - params.rr_y0.unsigned_abs() as usize
     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i8`, found `usize`

error[E0308]: mismatched types
    --> core/src/suggestions.rs:3483:5
     |
3482 | fn power(params: &HinterParams) -> usize {
     |                                    ----- expected `usize` because of return type
3483 |     params.rr_y1 - params.rr_y0.unsigned_abs() as usize
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `i8`
     |
help: you can convert an `i8` to a `usize` and panic if the converted value doesn't fit
     |
3483 |     (params.rr_y1 - params.rr_y0.unsigned_abs() as usize).try_into().unwrap()
     |     +                                                   +++++++++++++++++++++

error[E0277]: cannot subtract `usize` from `i8`
    --> core/src/suggestions.rs:3483:18
     |
3483 |     params.rr_y1 - params.rr_y0.unsigned_abs() as usize
     |                  ^ no implementation for `i8 - usize`
     |
     = help: the trait `std::ops::Sub<usize>` is not implemented for `i8`
     = help: the following other types implement trait `std::ops::Sub<Rhs>`:
               <&'a f32 as std::ops::Sub<f32>>
               <&'a f64 as std::ops::Sub<f64>>
               <&'a i128 as std::ops::Sub<i128>>
               <&'a i16 as std::ops::Sub<i16>>
               <&'a i32 as std::ops::Sub<i32>>
               <&'a i64 as std::ops::Sub<i64>>
               <&'a i8 as std::ops::Sub<i8>>
               <&'a isize as std::ops::Sub<isize>>
             and 48 others

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
Original diagnostics will follow.
```

As you can see, it incorrectly removed the parens, causing a compilation error.

The second issue was this one:

```
warning: redundant clone
   --> core/src/notifications.rs:484:34
    |
484 |         n.seen(n_disagree_myclass.clone(), 8);
    |                                  ^^^^^^^^ help: remove this
    |
    = note: `-D clippy::redundant-clone` implied by `-D warnings`
note: this value is dropped without further use
   --> core/src/notifications.rs:484:16
    |
484 |         n.seen(n_disagree_myclass.clone(), 8);
    |                ^^^^^^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
```

Which triggered this compilation error:

```
   --> core/src/notifications.rs:484:16
    |
470 |             (&n_disagree_myclass, vec![5, 6]),
    |              ------------------- borrow of `n_disagree_myclass` occurs here
...
484 |         n.seen(n_disagree_myclass, 8);
    |                ^^^^^^^^^^^^^^^^^^ move out of `n_disagree_myclass` occurs here
...
488 |         let mut expected_list = Vec::from(&expected_list_basic[1..]);
    |                                            ------------------- borrow later used here
```

In other words, the warning about a redundant clone was incorrect.

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```

### Additional Labels

* `l-suggestion-causes-error`
*

**Comment by giraffate** [comment]
Thanks for the report!

This has been fixed by https://github.com/rust-lang/rust-clippy/pull/9266, so I'm closing this.

**Comment by jgoerzen** [comment]
Thank you.  9266 looks relevant to the first issue, but I don't think it's relevant to the second (with the incorrect clone removal)

**Comment by Jarcho** [comment]
Can you give more context for the redundant_clone error.

**Comment by giraffate** [comment]
> The second issue was this one:

Oh, I missed it. Thanks for catching it.

**Comment by jgoerzen** [comment]
Sorry, I wasn't sure if these are related or separate.  Sounds like they are separate; my apologies for causing confusion with that.
